### ISSUE TITLE
How to create custom filter in Search Strategy?

### ISSUE THREAD

**Issue by chachus** [microsoft/nni#5441]
**Describe the issue**:
Since nn-meter is not working properly (maybe incompatibility with nni v2.10 model ir?) i was trying to create a custom model filter in the RegularizedEvolution strategy, but i cannot find a way to convert the model ir passed to the filter to something i can elaborate on (like corresponding PyTorch model). 
Thanks for the help.

**Environment**:
- NNI version: 2.10
- Training service (local|remote|pai|aml|etc): 
- Client OS: 
- Server OS (for remote mode only):
- Python version: 3.9
- PyTorch/TensorFlow version: 1.12
- Is conda/virtualenv/venv used?: yes
- Is running in Docker?: no

**How to reproduce it?**:

**Comment by matluster** [comment]
As of v2.10, I'm afraid there is no option other than copy-and-paste RegularizedEvolution and inject your logic into it.

As of latest (development) version, custom filter has already been supported. See https://github.com/microsoft/nni/blob/master/nni/nas/strategy/middleware.py

**Comment by chachus** [comment]
I undertand, thanks for the reply.
In that case how can i manipulate the model. Because when i look at the model obtained from 'get_targeted_model()" i obtain a frozen model with an id and other informations, with no mutation applied. 

`Model(model_id=7145, status=ModelStatus.Frozen, graphs=['_model'], evaluator=FunctionalEvaluator(_nni_symbol=<class 'nni.nas.evaluator.functional.FunctionalEvaluator'>, _nni_args=[], _nni_kwargs={'function': <function compute_zico_score at 0x7fa968ae>, 'train_data': <torch.utils.data.dataloader.DataLoader object at 0x7fa960>, 'max_batch': 2, 'lossfunc': CrossEntropyLoss(), 'gpu': True}, _nni_call_super=True, function=<function compute_zico_score at 0x7fa968ae0>, arguments={'train_data': <torch.utils.data.dataloader.DataLoader object at 0x7fa96278>, 'max_batch': 2, 'lossfunc': CrossEntropyLoss(), 'gpu': True}), metric=None, intermediate_metrics=[], python_class=<class 'model_space.ModelSpace'>`

It is not clear to me how i can obtain a model i can work with.

Eventually the development version is in an usable state?

**Comment by matluster** [comment]
Sorry for the late reply. I think I've missed some notifications.

> manipulate the model.

Do you mean modify a frozen model? This is generally not recommended. Rather, we recommend re-freeze a model from the model space.

> Eventually the development version is in an usable state?

I think it's on its way of becoming "stabilized".

**Comment by chachus** [comment]
No problem, Thanks a lot for the reply.

> Do you mean modify a frozen model?
Basically my intention was to filter models based on some parameters (e.g. #Parameters, FLOPS, Latency) before submitting them for evaluation. To do this i need to access the model with all choices made from the NAS Strategy, which is not the case when using the Callable model filter parameter in the RegularizedEvolution Strategy, where i receive this: 
`Model(model_id=7145, status=ModelStatus.Frozen, graphs=['_model'], evaluator=FunctionalEvaluator(_nni_symbol=<class 'nni.nas.evaluator.functional.FunctionalEvaluator'>, _nni_args=[], _nni_kwargs={'function': <function compute_zico_score at 0x7fa968ae>, 'train_data': <torch.utils.data.dataloader.DataLoader object at 0x7fa960>, 'max_batch': 2, 'lossfunc': CrossEntropyLoss(), 'gpu': True}, _nni_call_super=True, function=<function compute_zico_score at 0x7fa968ae0>, arguments={'train_data': <torch.utils.data.dataloader.DataLoader object at 0x7fa96278>, 'max_batch': 2, 'lossfunc': CrossEntropyLoss(), 'gpu': True}), metric=None, intermediate_metrics=[], python_class=<class 'model_space.ModelSpace'>`

> I think it's on its way of becoming "stabilized".
I will look forward to it! :)
